File: Structural/Bridge/Cpp/TV.cpp
// TV.cpp

#include "TV.h"

void TV::turnOn() {
  on = true;
  std::cout << "TV is now ON." << std::endl;
}

void TV::turnOff() {
  on = false;
  std::cout << "TV is now OFF." << std::endl;
}

void TV::setVolume(int volume) {
  this->volume = volume;
  std::cout << "TV volume set to " << volume << std::endl;
}

int TV::getVolume() const {
  return volume;
}

bool TV::isOn() const {
  return on;
}

File: Structural/Bridge/CSharp/TV.cs
// TV.cs

using System;

public class TV : IDevice {
  private bool isOn = false;
  private int volume = 50;

  public void TurnOn() {
    isOn = true;
    Console.WriteLine("TV is now ON.");
  }

  public void TurnOff() {
    isOn = false;
    Console.WriteLine("TV is now OFF.");
  }

  public void SetVolume(int volume) {
    this.volume = volume;
    Console.WriteLine($"TV volume set to {volume}");
  }

  public int GetVolume() {
    return volume;
  }

  public bool IsOn() {
    return isOn;
  }
}
